spec/features/user_authentication_with_cis2_spec.rb

```diff
@@ -97,6 +109,43 @@ def cis2_auth_mock
     and_i_am_logged_in
   end
 
+  scenario "user has wrong team selected" do
+    setup_cis2_auth_mock
+
+    given_the_cis2_feature_flag_is_enabled
+    when_i_go_to_the_sessions_page
+    then_i_am_on_the_start_page
+
+    when_i_click_the_cis2_login_button
+    then_i_see_the_team_not_found_error
+
+    given_my_team_is_setup_in_mavis
+    when_i_click_the_change_role_button
+    then_i_see_the_sessions_page
+  end
+
+  context "user has no other orgs to select" do
+    let(:cis2_auth_mock) do
+      super().tap do |mock|
+        mock["extra"]["raw_info"]["nhsid_nrbac_roles"].select! do
+          _1["org_code"] == test_team_ods_code
+        end
+      end
+    end
+
+    scenario "user has wrong team selected" do
+      setup_cis2_auth_mock
+
+      given_the_cis2_feature_flag_is_enabled
+      when_i_go_to_the_start_page
+      then_i_should_see_the_cis2_login_button
+
+      when_i_click_the_cis2_login_button
+      then_i_see_the_team_not_found_error
+      and_there_is_no_change_role_button
+    end
+  end
```

Comment:
I don't think we should have 3 scenarios in one feature spec. I think we should split these up into separate tests... It's probably more effort rewriting some of the steps but for consistency we try to stick to 1 scenario per file in all except for one case

Reply:
I've gone through and split this up into 3 files, each with a couple of scenarios. I'm not sure being really strict about only one scenario per file is right ... it becomes quite tedious to name the files and thinking about how they fit together, imo. Now that I've split this up, I actually think that having all the CIS2 auth stuff in one file was alright, but I think this is a reasonable middle-ground too.

Let me know what you think.

Reply:
Yeah, I think this feels a bit better... let's see how it works out as we iterate